Sources/ProjectDescription/MergedBinaryType.swift

```diff
@@ -0,0 +1,9 @@
+/// Represents the different options to configure a target for mergeable libraries
+public enum MergedBinaryType: Equatable, Codable {
+    /// Target is never going to merge available dependencies
+    case disabled
+    /// Target is going to merge direct target dependencies (just the ones declared as part of it's project)
+    case automatic
+    /// Target is going to merge direct and specified dependencies that are not part of the project.
```

Comment:
Can you elaborate on what this means?
Also, given `manual` with empty set is the same as `automatic`, I don't think we need 2 separate cases

Reply:
Added some information based on Apple's official one and adding link to the public doc

Reply:
I'm sorry that I'm not super familiar with how mergeable libraries work 🙏 

If I understand correctly:
- automatic -> only direct dependencies
- manual -> direct dependencies plus the one you specify

If I got it correctly, `.manual([])` is the same as `automatic`, so we don't need both.
Otherwise, I think we should make the documentation clearer.

Reply:
The first point is correct but let me clarify what a direct dependency is (I had the same doubt, this is added as part of the doc).

A direct dependency is not every single dependency your target depends on. It's a dependency that belongs to your project (as a target) and is linked (not embedded). This is going to exclude all precompiled libraries

The difference between automatic and manual resides in the way Xcode considers what to merge.

In the case of automatic, it is considering all dynamic library targets the merging target is depending on.

In the case of manual it is considering all dependencies marked with MAKE_MERGEABLE and MERGEABLE_LIBRARY enabled. And for precompiled libraries the ones added on OTHER_LDFLAGS with the flags added on DefaultSettingsProvider.Swift

So they are not quite the same and adding the difference allows you to follow Apple's doc easily.

Reply:
Oh, I missed that it was setting a different `MERGED_BINARY_TYPE` in the settings, makes sense now